### PR TITLE
Play Melody

### PR DESCRIPTION
**Pull request by hexthat**
add tiny feature

### CODE REVIEW

File: adafruit_circuitplayground/express.py [hunk 1]
```diff
@@ -658,10 +658,41 @@ def stop_tone(self):
...
+            587,659,698,784,880,988,1047,988,659,
+            784,698,880,784,988,880,1047,988,1175,1047,1319,1175,1397,1319,1319,1397,1175,1319,
+            0,0,0]
+
+            beats= [2,2,4,2,2,4,2,2,2,2,4,2,2,2,2,8,2,2,2,10,
+            2,2,4,2,2,4,2,2,2,2,4,2,2,2,2,8,2,2,2,10,
+            2,2,4,2,2,4,2,2,8,2,2,4,2,2,4,2,2,8,
+            2,2,4,2,2,4,2,2,8,
+            2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,2,6,
+            4,2,2]
```
Comment (sommersoft) [inline adafruit_circuitplayground/express.py]: pylint is complaining about the line length (>100) on line 676, and several instances of a trailing whitespace. Beyond that, the indenture is wrong for each list; the continuation lines should be indented to the opening bracket at minimum, when the items start on the same line as the `[`. To maximize the line lenghts, i would suggest using this style (which also meets PEP8's 79 char line length):
```python
           notes = [
               698, 880, 988, 698, 880, 988, 698, 880, 988, 1319, 1175, 988, 1047, 988,
               784, 659, 587, 659, 784, 659, 698, 880, 988, 698, 880, 988, 698, 880, 988,
               1319, 1175, 988, 1047, 1319, 988, 784, 988, 784, 587, 659, 587,659,698,784,
               880, 988, 1047, 988, 659, 698, 784, 880, 988, 1047, 1175, 1319, 1397, 1568,
               587, 659, 698, 784, 880, 988, 1047, 988, 659, 784, 698, 880, 784, 988, 880,
               1047, 988, 1175, 1047, 1319, 1175, 1397, 1319, 1319, 1397, 1175, 1319, 0,
               0, 0
           ]

           beats = [
               2, 2, 4, 2, 2, 4, 2, 2, 2, 2, 4, 2, 2, 2, 2, 8, 2, 2, 2, 10,
               2, 2, 4, 2, 2, 4, 2, 2, 2, 2, 4, 2, 2, 2, 2, 8, 2, 2, 2, 10,
               2, 2, 4, 2, 2, 4, 2, 2, 8, 2, 2, 4, 2, 2, 4, 2, 2, 8, 2, 2,
               4, 2, 2, 4, 2, 2, 8, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
               2, 1, 1, 2, 6, 4, 2, 2
           ]
```
Oh, and spaces between comma separated values increase readability.

File: adafruit_circuitplayground/express.py [hunk 2]
```diff
@@ -658,10 +658,41 @@ def stop_tone(self):
...
+
+            while True:
+                cpx.play_melody(notes, beats)
+        """        
+        for i in range(len(frequencies)):
```
Comment (sommersoft) [inline adafruit_circuitplayground/express.py]: pylint is suggesting the use of `enumerate()` here. I agree. Would look something like this:

```python
for idx, frequency in enumerate(frequencies):
    self.play_tone(frequency, durations[idx]/8)
```
Also, I would add a conditional to verify that the supplied lenghts of `frequencies` and `durations` are equal, and raise an exception if they are not. Makes it easier for the user to debug, as opposed to having an `IndexError` raised after the melody has started playing.

File: adafruit_circuitplayground/express.py [hunk 3]
```diff
@@ -658,10 +658,41 @@ def stop_tone(self):
         # Stop playing any tones.
         if self._sample is not None and self._sample.playing:
             self._sample.stop()
-            self._sample.deinit()
-            self._sample = None
```
Comment (sommersoft) [inline adafruit_circuitplayground/express.py]: Why are these two lines deleted?

Comment (hexthat) [reply adafruit_circuitplayground/express.py]: undesired pause added at the end of tones

should i add back a command to deinit() the pin? it is useful but shouldnt be called willy nilly

**Comment by hexthat** [comment]
I changed it a bit, it think this is a nicer version?

**Comment by hexthat** [comment]
i can read travis fails now

File: adafruit_circuitplayground/express.py [hunk 4]
```diff
@@ -662,6 +662,52 @@ def stop_tone(self):
...
+        """
+        # Play a melody.
+        for idx, frequency in enumerate(frequencies):
+            if frequency is 0:
+                self._sample.stop()
+                time.sleep(durations[idx]/speed)
+            else:
+                self.start_tone(frequency)
+                time.sleep(durations[idx]/speed)
+                self._sample.stop()
```
Comment (sommersoft) [inline adafruit_circuitplayground/express.py]: While `is 0` works, I would think equality comparison is better than membership comparison here. Also, if `self._sample.stop()` is always called no matter the `frequency`, you could shorten this.
```python
for idx, frequency in enumerate(frequencies):
    if frequency != 0:
        self.start_tone(frequency)
    time.sleep(durations[idx]/speed)
    self._sample.stop()
self._sample.stop()

Comment (hexthat) [reply adafruit_circuitplayground/express.py]: yes it is needed, if you dont call self._sample.stop() before self.start_tone() the new tone will not play

so you would like something like this instead?

```
for idx, frequency in enumerate(frequencies):
    if frequency != 0:
        self._sample.stop()
        self.start_tone(frequency)
    self._sample.stop()
    time.sleep(durations[idx]/speed)
    self._sample.stop()
self.stop_tone()
```

Comment (hexthat) [reply adafruit_circuitplayground/express.py]: yes this works very well thanks.

File: adafruit_circuitplayground/express.py [hunk 5]
```diff
@@ -569,9 +562,10 @@ def _sine_sample(length):
         for i in range(length):
             yield int(tone_volume * math.sin(2*math.pi*(i / length)) + shift)
 
-    def _generate_sample(self, length=100):
+    def _generate_sample(self):
```
Comment (sommersoft) [inline adafruit_circuitplayground/express.py]: `length` should remain a kwarg, and not be hard-coded in `_generate_sample()`. In `start_tone()`, length could be a number other than 100. Starting at line # 625:
```python
self._speaker_enable.value = True
length = 100
if length * frequency > 350000:
    length = 350000 // frequency
self._generate_sample(length)
```

Comment (sommersoft) [reply adafruit_circuitplayground/express.py]: I may need more coffee today. I see that you removed that section in `start_tone()`, but I still don't understand the why behind it.

Comment (hexthat) [reply adafruit_circuitplayground/express.py]: to remove length param from _generateh_sample() when need to remove it from _sine_sample() too, then add the math for it to _sine_sample() super easy just '100'...

Comment (hexthat) [reply adafruit_circuitplayground/express.py]: dargh nvm i see what your saying i mistakenly removed it for no reason, even though its not used ever

**Comment by hexthat** [comment]
```
from adafruit_circuitplayground.express import cpx
import time

def play_melody(frequencies, durations, speed=10):
    for idx, frequency in enumerate(frequencies):
        if frequency != 0:
            cpx.start_tone(frequency)
        time.sleep(durations[idx]/speed)
        cpx._sample.stop()
    cpx.stop_tone()

notes = [293, 440, 440, 0, 330, 494, 349, 523, 523, 0, 330, 494,
293, 440, 440, 0, 330, 494, 349, 523, 523, 0, 330, 494,
293, 349, 587, 293, 349, 587,
659, 698, 659, 698, 659, 523, 440,
440, 293, 349, 392, 440, 440, 293, 349, 392, 330,
293, 349, 587, 293, 349, 587,
659, 698, 659, 698, 659, 523, 440,
440, 293, 349, 392, 440,
440, 293]

beats = [2, 2, 2, 1, 1, 4, 2, 2, 2, 1, 1, 4,
2, 2, 2, 1, 1, 4, 2, 2, 2, 1, 1, 4,
1, 1, 4, 1, 1, 4,
3, 1, 1, 1, 1, 1, 4, 2, 2, 1, 1, 6,
2, 2, 1, 1, 6,
1, 1, 4, 1, 1, 4,
3, 1, 1, 1, 1, 1, 4, 2, 2, 1, 1, 6,
2, 6]

while True:
    play_melody(notes, beats)
    time.sleep(2)
```

no need to change frozen to play melody

**Pull request closed without merging**